pkg/storage/stores/shipper/uploads/table.go

```diff
@@ -226,15 +226,18 @@ func (lt *Table) MultiQueries(ctx context.Context, queries []chunk.IndexQuery, c
 }
 
 func (lt *Table) getOrAddDB(name string) (*bbolt.DB, error) {
+	lt.dbsMtx.RLock()
+	db, ok := lt.dbs[name]
+	lt.dbsMtx.RUnlock()
+
+	if ok {
+		return db, nil
+	}
+
 	lt.dbsMtx.Lock()
 	defer lt.dbsMtx.Unlock()
 
-	var (
-		db  *bbolt.DB
-		err error
-		ok  bool
-	)
-
+	var err error
 	db, ok = lt.dbs[name]
 	if !ok {
```

Comment:
I might invert this `if` so that the read-lock and write-lock cases look the same.  Also this would outdent the remainder of the function which is more idiomatic Go.

Reply:
Sorry, I merged the PR already. Will open a follow-up PR.